src/dask_histogram/boost.py

```diff
@@ -249,6 +249,10 @@ def fill(  # type: ignore
         else:
             raise ValueError(f"Cannot interpret input data: {args}")
 
+        # always regenerate the local histref before sending off (so copies, name assignment work)
+        new_histref = (tuple(self.axes), self.storage_type(), self.metadata)
+        self._histref = new_histref
```

Comment:
Surprised to see this in this method. So updating the .axes doesn't have an effect until calling fill().
In fact, it makes it feel like `_histref` is a property not an attribute.

Reply:
Yeah the problem is that since the dask_histogram object is itself a histogram with state, if we manipulate that then we have to update the `_histref` so anything we do to the histogram locally that matters is propagated to the cluster. 
We cannot just use self as metadata, it is too heavy serialization wise.

Reply:
OK - I have addressed this, and `_histref` is now a property as opposed to an attribute!